partition_filter.go

```diff
@@ -67,6 +69,38 @@ func NewPartitionFilterByKey(key *Key) *PartitionFilter {
 	return &PartitionFilter{Begin: key.PartitionId(), Count: 1, Digest: key.Digest()}
 }
 
+func NewPartitionFilterSelectPartitions(partitionIds []int) (*PartitionFilter, error) {
+	minPartitionId := _PARTITIONS
+	maxPartitionId := 0
+
+	// Need to find min and max partition ids to know what the range or partitions is
+	for _, id := range partitionIds {
+		if id < 0 || id >= _PARTITIONS {
+			return nil, newError(types.PARAMETER_ERROR, "Partition id out of range: %d", strconv.Itoa(id))
+		}
+
+		minPartitionId = min(minPartitionId, id)
+		maxPartitionId = max(maxPartitionId, id)
+	}
+	partitionCount := maxPartitionId - minPartitionId
+	partitionFilter := newPartitionFilter(minPartitionId, partitionCount)
+	slices.Sort(partitionIds)
```

Comment:
Before sorting, this slice should be copied. Slices are pointers and the user owns this slice.